### ISSUE TITLE
Docstrings in outputs cause nbconvert to fail

### ISSUE THREAD

**Issue by timkpaine** [jupyterlab/jupyterlab#4447]
Given the following single cell notebook, convert to pdf fails:

<img width="687" alt="screen shot 2018-04-24 at 3 38 55 pm" src="https://user-images.githubusercontent.com/3105306/39210076-a74244e8-47d5-11e8-88ac-ed74bd098338.png">

<img width="675" alt="screen shot 2018-04-24 at 3 39 26 pm" src="https://user-images.githubusercontent.com/3105306/39210105-bc5188d0-47d5-11e8-8529-d250e2598d25.png">

error:

```
! Text line contains an invalid character.
l.291 ^^[
         [0;31mInit signature:^^[[0m ^^[[0mdatetime^^[[0m^^[[0;34m.^^[[0m^^[...
```

Confirmed with texlive on mac and centos. Can't repro on classic notebook as it didn't output docstrings to the output area of the cell.

**Comment by ian-r-rose** [comment]
Is it possible to try running nbconvert with `xelatex`? I think `pdflatex` chokes on may unicode characters.

**Comment by timkpaine** [comment]
@ian-r-rose is that a q for me? this is just what nbconvert does by default, it should've already been using that: http://nbconvert.readthedocs.io/en/latest/changelog.html?highlight=xelatex#id8

**Comment by timkpaine** [comment]
ah, I had scrolled in the output, it was already using that:
```
nbconvert failed: PDF creating failed, captured latex output:
This is XeTeX, Version 3.14159265-2.6-0.99992 (TeX Live 2015) (preloaded format=xelatex)
 restricted \write18 enabled.
entering extended mode
(/Users/theocean154/Programs/projects/notebook.tex
LaTeX2e <2015/01/01>
Babel <3.9l> and hyphenation patterns for 79 languages loaded.
(/usr/local/texlive/2015/texmf-dist/tex/latex/base/article.cls
Document Class: article 2014/09/29 v1.4h Standard LaTeX document class
(/usr/local/texlive/2015/texmf-dist/tex/latex/base/size11.clo))
(/usr/local/texlive/2015/texmf-dist/tex/latex/base/fontenc.sty
(/usr/local/texlive/2015/texmf-dist/tex/latex/base/t1enc.def))
(/usr/local/texlive/2015/texmf-dist/tex/latex/psnfss/mathpazo.sty)
(/usr/local/texlive/2015/texmf-dist/tex/latex/graphics/graphicx.sty
(/usr/local/texlive/2015/texmf-dist/tex/latex/graphics/keyval.sty)
(/usr/local/texlive/2015/texmf-dist/tex/latex/graphics/graphics.sty
```

**Comment by ian-r-rose** [comment]
Yeah, it should have been the default, just double checking.

**Comment by timkpaine** [comment]
[notebook.tex.txt](https://github.com/jupyterlab/jupyterlab/files/1944371/notebook.tex.txt)

**Comment by jasongrout** [comment]
I'll just point out that the characters it is complaining about are the ASCII 1B "ESC" characters preceding the `[`:

```
xxd -u notebook.tex
[snip]
00003bc0: 5C62 6567 696E 7B76 6572 6261 7469 6D7D  \begin{verbatim}
00003bd0: 0A1B 5B30 3B33 316D 496E 6974 2073 6967  ..[0;31mInit sig
00003be0: 6E61 7475 7265 3A1B 5B30 6D20 1B5B 306D  nature:.[0m .[0m
00003bf0: 6461 7465 7469 6D65 1B5B 306D 1B5B 303B  datetime.[0m.[0;
00003c00: 3334 6D2E 1B5B 306D 1B5B 306D 6461 7465  34m..[0m.[0mdate
00003c10: 1B5B 306D 1B5B 303B 3334 6D28 1B5B 306D  .[0m.[0;34m(.[0m
00003c20: 1B5B 306D 7365 6C66 1B5B 306D 1B5B 303B  .[0mself.[0m.[0;
```

**Comment by vidartf** [comment]
I.e. ANSI terminal color escape codes.

**Comment by vidartf** [comment]
Is it possible that the template is missing a `| ansi2latex` filter somewhere?

**Comment by telamonian** [comment]
This is fixed by jupyter/nbconvert#1181

> Is it possible that the template is missing a `| ansi2latex` filter somewhere?

@vidartf Yeah pretty much
